pkg/worker/worker.go

```diff
@@ -42,19 +42,39 @@ const (
 	SonobuoyNSEnvKey              = "SONOBUOY_NS"
 	SonobuoyPluginPodEnvKey       = "SONOBUOY_PLUGIN_POD"
 	SonobuoyWorkerContainerEnvKey = "SONOBUOY_WORKER_CONTAINER"
+
+	// defaultWaitFileConsumptionDelay provides a minimum time window where sidecars may intercept results and remove
+	// the done file before Sonobuoy decides to process it.
+	defaultWaitFileConsumptionDelay = 5 * time.Second
+
+	// SonobuoyDoneFileDelayKey is the env key specifying where this value may be overwritten. Adding this in as a mitigation for problems
+	// caused by adding the delay. I anticipate we will not have a problem though since the default delay is reasonably short.
+	SonobuoyDoneFileDelayKey = "SONOBUOY_DONE_FILE_DELAY"
 )
 
 var (
 	// apiClient is used to check the status of this pod and other containers in it. If it is
 	// nil, those checks will be skipped.
 	apiClient kubernetes.Interface
+
+	// waitFileConsumptionDelay provides a minimum time window where sidecars may intercept results and remove
+	// the done file before Sonobuoy decides to process it.
+	waitFileConsumptionDelay time.Duration
 )
 
 func init() {
 	err := mime.AddExtensionType(".gz", "application/gzip")
 	if err != nil {
 		logrus.Error(err)
 	}
+
+	if v := os.Getenv(SonobuoyDoneFileDelayKey); len(v) > 0 {
```

Comment:
But how would someone actually modify this? In theory it could be overwritten, but we don't expose the worker env to the users and dont have any flags for that currently.

Reply:
I think that this just requires a small extension to the way we handle the env vars. Whenever we make the sonobuoy worker, perhaps we just copy over the env from the plugin itself. That way its easy for the user to set those values and it allows for per-plugin values rather than some global 'worker env' we set.

Reply:
We currently dont have the means to do that (the createworkercontainerdefintion method doesn't have access to that nor does the plugin interface).

So the change is _slightly_ non-trivial so I'm going to put that in a separate PR. We just have to have a separate call to add env values to the worker image.